### PR TITLE
Adding diagnostic message when using duplicate arguments in function …

### PR DESCRIPTION
**Pull request by CTrando**
…definition.

```
def test(a, a=1, *a, **a):
  pass
```

The above code gives a syntax error. Same as pylint duplicate-argument-name (E0108).

### CODE REVIEW

File: src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs [hunk 1]
```diff
@@ -367,6 +374,20 @@ private bool EvaluateFunctionBody(IPythonFunctionType fn)
             return parameters;
         }
 
+        private void ValidateParameters(FunctionDefinition fd) {
+            var duplicates = fd.Parameters.Where(p => !string.IsNullOrEmpty(p.Name)).GroupBy(p => p.Name).Where(g => g.Count() > 1).Select(y => y.Key);
```
Comment (CTrando) [inline src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs]: Could anyone give me a nicer syntax for this? 
I'm trying to find all duplicate names in ```fd.Parameters```

Comment (jakebailey) [reply src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs]: I don't think there's a nicer way off the top of my head. Maybe `fd.Parameters.GroupBy(p => p.Name).SelectMany(g => g.Skip(1)).Distinct()`, but I'm not convinced that's better.

Comment (CTrando) [reply src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs]: yeah and we have to add the string null empty check on top of that too, so maybe this is the best.

File: src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs [hunk 2]
```diff
@@ -367,6 +374,20 @@ private bool EvaluateFunctionBody(IPythonFunctionType fn)
...
+
+            foreach(var paramName in duplicates) {
```
Comment (jakebailey) [inline src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Callables.cs]: Nit: formatting (`foreach (var ...`)

**Comment by jakebailey** [comment]
Fix the merge conflicts and this is fine.

**Pull request merged**